src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -140,6 +148,166 @@ public void canPreventSQLInjection() throws IOException, SQLException, InvalidNa
+        fareRuleInput.contains_id = "any";
+        fareRuleInput.origin_id = "value";
+        fareRuleInput.destination_id = "permitted";
+        fareInput.fare_rules = new FareRuleDTO[]{fareRuleInput};
+
+        // convert object to json and save it
+        JdbcTableWriter createTableWriter = createTestTableWriter(fareTable);
+        String createOutput = createTableWriter.create(mapper.writeValueAsString(fareInput), true);
+        LOG.info("create {} output:", fareTable.name);
+        LOG.info(createOutput);
+
+        // parse output
+        FareDTO createdFare = mapper.readValue(createOutput, fareDTOClass);
+
+        // make sure saved data matches expected data
+        assertThat(createdFare.fare_id, equalTo(fareId));
+        assertThat(createdFare.fare_rules[0].fare_id, equalTo(fareId));
+
+        // try to update record
+        String updatedFareId = "3B";
+        createdFare.fare_id = updatedFareId;
+
+        // covert object to json and save it
+        JdbcTableWriter updateTableWriter = createTestTableWriter(fareTable);
+        String updateOutput = updateTableWriter.update(
+                createdFare.id,
+                mapper.writeValueAsString(createdFare),
+                true
+        );
+        LOG.info("update {} output:", fareTable.name);
+        LOG.info(updateOutput);
+
+        FareDTO updatedFareDTO = mapper.readValue(updateOutput, fareDTOClass);
+
+        // make sure saved data matches expected data
+        assertThat(updatedFareDTO.fare_id, equalTo(updatedFareId));
+        assertThat(updatedFareDTO.fare_rules[0].fare_id, equalTo(updatedFareId));
+
+        // try to delete record
+        JdbcTableWriter deleteTableWriter = createTestTableWriter(fareTable);
+        int deleteOutput = deleteTableWriter.delete(
+                createdFare.id,
+                true
+        );
+        LOG.info("delete {} output:", fareTable.name);
+        LOG.info(updateOutput);
+
+        // make sure fare_attributes record does not exist in DB
+        String sql = String.format(
+                "select * from %s.%s where id=%d",
+                testNamespace,
+                fareTable.name,
+                createdFare.id
+        );
+        LOG.info(sql);
+        ResultSet rs = testDataSource.getConnection().prepareStatement(sql).executeQuery();
+        assertThat(rs.getFetchSize(), equalTo(0));
+
+        // make sure fare_rules record does not exist in DB
+        String fareRulesSql = String.format(
+                "select * from %s.%s where id=%d",
+                testNamespace,
+                Table.FARE_RULES.name,
+                createdFare.fare_rules[0].id
+        );
+        LOG.info(fareRulesSql);
+        ResultSet fareRulesResultSet = testDataSource.getConnection().prepareStatement(fareRulesSql).executeQuery();
+        assertThat(fareRulesResultSet.getFetchSize(), equalTo(0));
```

Comment:
Now that I look at these tests, it seems like they could really use a sql check to make sure that the data was created and updated.  Also, these 2 lines could be made into a reusable method `assertThatSqlQueryYieldsZeroRows` or something like that.

Reply:
Good idea.

Reply:
Hmm, actually I think the checking for created/updated data gets a little muddied because we would then need to ensure that we're using the same connection to see the data we expect. I think the first part of your comment is out of scope.